### ISSUE TITLE
Focus problem at mobile controlled component

### ISSUE THREAD

**Issue by jandramila** [dreyescat/react-rating#24]
So in my use case, i have a Rating component where i show the average score for a given product, and whenever a user clicks on a specific rate, a prompt to leave comments opens with another Rating component showing the given rate, BUT the first Rating component should go back to the average score.

While checking through closed issues i found #16 so i took those lines to set up my desired behaviour resulting in this:

![Desktop Behaviour](http://g.recordit.co/99Lk7Y2gpz.gif)

Which works exactly as it is supposed to, but on mobile, the first Rating component keeps showing the last clicked rating until i click somewhere else:

![Mobile Behaviour](http://g.recordit.co/iKckpTGhzN.gif)

There's no activeElement after the onChange, i´ve forked the fiddle on the mentioned (#16) issue to replicate this behaviour and it behaves the same, works on desktop, not on mobile:

https://jsfiddle.net/jandramila/k11mcb8v/3/

Any thoughts or workarounds?

**Comment by dreyescat** [comment]
Such a nice bug report 👍. It makes you want to help!

I don't have the right tools now to look into it but maybe it is related to how mouse/tap events are handled in touch devices (mobiles basically).

Just to see if I am on the same page...

The forked jsfiddle on a desktop does not set the rating when you click on a symbol because you set the rate to `undefined` at the end of the `onChange` handler:

```javascript
onChange: function (rate) {
  this.setState({rate: rate});
  this.setState({rate: undefined});
},
```

However, if you run it on a mobile it looks like the second `setState`, the one that sets the rate to `undefined`, is ignored, and the symbol remains selected.

Is this the behaviour you are trying to reproduce with this jsfiddle?

**Comment by jandramila** [comment]
Exactly, i thought so, it seems like the :hover status remains still because of tap (expected) behaviour, setting my stars on the rate i selected. I tested it using mobile mode with chrome devtools, it's just that the recorder i used to make the GIFs awalys shows the default pointer instead of the 'tap' kind.

If you can't check this right now, there's no problem at all, a work mate was interested in giving it a try so we might take a look at it, but most surely by next week. 

Thanks for your quick response!

**Comment by dreyescat** [comment]
@jandramila I think I found the problem. Could you please confirm it is solved in [v0.6.0](https://github.com/dreyescat/react-rating/releases/tag/v0.6.0)?

**Comment by jandramila** [comment]
Problem solved, thank you very much!
